data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -30,12 +34,21 @@ public Results execute(String database, String command) throws Exception {
         return execute(database, command, clusterEndpoint);
     }
 
-    private Results execute(String database, String command, String clusterEndpoint) throws Exception {
-        String aadAccessToken = aadAuthenticationHelper.acquireAccessToken();
+    private Results execute(String database, String command, String clusterEndpoint) throws DataServiceException, DataClientException {
+        // Argument validation:
+        if(StringUtils.isAnyEmpty(database, command, clusterEndpoint)){
+            throw new IllegalArgumentException("database, command or clusterEndpoint are empty");
+        }
 
-        String jsonString = new JSONObject()
+        String aadAccessToken = aadAuthenticationHelper.acquireAccessToken();
+        String jsonString;
+        try {
+            jsonString = new JSONObject()
                 .put("db", database)
                 .put("csl", command).toString();
+        } catch (JSONException e) {
+            throw new DataClientException(clusterEndpoint, String.format(clusterEndpoint, "Error in executing command: %s, in database: %s", command, database), e);
+        }
 
         return Utils.post(clusterEndpoint, aadAccessToken, jsonString);
```

Comment:
Is aadAccessToken the only header we send?
Can you please add an issue to send all relevant headers as they appear [here](https://github.com/Azure/azure-kusto-python/blob/4271142df391f526d241d96b691c3b5f3a0ed10d/azure-kusto-data/azure/kusto/data/request.py#L293)?

Reply:
It's not the only header. Currently, we are sending those headers:

        httpPost.addHeader("Authorization", String.format("Bearer %s", aadAccessToken));
        httpPost.addHeader("Content-Type", "application/json");
        httpPost.addHeader("Accept-Encoding", "gzip,deflate");
        httpPost.addHeader("Fed", "True");
        httpPost.addHeader("x-ms-client-version", "Kusto.Java.Client");

Reply:
It's not the only header. Currently, we are sending those headers:

        httpPost.addHeader("Authorization", String.format("Bearer %s", aadAccessToken));
        httpPost.addHeader("Content-Type", "application/json");
        httpPost.addHeader("Accept-Encoding", "gzip,deflate");
        httpPost.addHeader("Fed", "True");
        httpPost.addHeader("x-ms-client-version", "Kusto.Java.Client");